### PR TITLE
Feat/demo home coordinator

### PR DESCRIPTION
**Pull request by aleksgeyman**

### CODE REVIEW

**Review by veselaIl — APPROVED** [review]

File: SFBaseKitDemo/SFBaseKitDemo/AppCoordinator/AppCoordinator.swift [hunk 1]
```diff
@@ -24,11 +23,11 @@ class AppCoordinator: Coordinator {
         self.window?.makeKeyAndVisible()
         
         let loginCoordinator = LoginCoordinator(navigationController: navigationController)
+        loginCoordinator.parentCoordinator = self
```
Comment (dvp-petrov) [inline SFBaseKitDemo/SFBaseKitDemo/AppCoordinator/AppCoordinator.swift]: addChildCoordinator automatically sets parentCoordinator. This line is not needed.

File: SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+//  SFBaseKitDemo
+//
+//  Created by Aleksandar Geyman on 21.04.20.
+//  Copyright © 2020 Upnetix. All rights reserved.
+//
+
+import SFBaseKit
+
+protocol HomeSceneDelegate: Coordinator {
+
```
Comment (dvp-petrov) [inline SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeCoordinator.swift]: New functionality request - Add logout method in this protocol.

File: SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeCoordinator.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    unowned private var navController: UINavigationController!
+    
+    override func start() {
+        
+        // Instantiate Home Screen and add scene delegate.
+        let rootVC = HomeViewController.instantiateFromStoryboard()
+        rootVC?.sceneDelegate = self
+        
+        // As login flow is finished new navigation flow is started.
+        let navController = UINavigationController(rootViewController: rootVC ?? UIViewController())
```
Comment (dvp-petrov) [inline SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeCoordinator.swift]: We discussed that - line 25 to 29 should be moved inside of AppCooridnator.

File: SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeViewController.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+//
+//  Created by Aleksandar Geyman on 21.04.20.
+//  Copyright © 2020 Upnetix. All rights reserved.
+//
+
+import SFBaseKit
+
+class HomeViewController: UIViewController {
+    
+    unowned var sceneDelegate: HomeSceneDelegate!
```
Comment (dvp-petrov) [inline SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeViewController.swift:17]: Move HomSceneDelegate protocol to be defined in this class. (this is the rule we discussed on the weekly meeting)

**Review by dvp-petrov — CHANGES_REQUESTED** [review]

File: SFBaseKitDemo/SFBaseKitDemo/AppCoordinator/AppCoordinator.swift (hunk 1 above)
Comment (aleksgeyman) [reply SFBaseKitDemo/SFBaseKitDemo/AppCoordinator/AppCoordinator.swift]: fixed

File: SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeViewController.swift (hunk 4 above)
Comment (aleksgeyman) [reply SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeViewController.swift:17]: I've managed to watch the recording. Moved the protocol in the view controller file.

File: SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeCoordinator.swift (hunk 2 above)
Comment (aleksgeyman) [reply SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeCoordinator.swift]: Done.

File: SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeCoordinator.swift (hunk 3 above)
Comment (aleksgeyman) [reply SFBaseKitDemo/SFBaseKitDemo/ExampleHomeModule/HomeCoordinator.swift]: Done.

**Review by veselaIl — APPROVED** [review]

**Review by dvp-petrov — APPROVED** [review]

**Pull request merged**